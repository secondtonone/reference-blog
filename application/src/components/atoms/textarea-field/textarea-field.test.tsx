import renderer from 'react-test-renderer';
import Textarea from './textarea-field';

jest.mock('../tooltip', () => ({ children }) => <span>{children}</span>);
const testHandler = () => ({});

describe('Match snapshots Textarea', () => {
  it('empty field', () => {
    const tree = renderer.create(<Textarea placeholder="Text" value="" onChange={testHandler} scrollTooltip={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with value', () => {
    const tree = renderer.create(<Textarea value="Text" onChange={testHandler} scrollTooltip={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('internal', () => {
    const tree = renderer.create(<Textarea value="Text" internal onChange={testHandler} scrollTooltip={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('with error', () => {
    const tree = renderer.create(<Textarea value="Text" onChange={testHandler} state="invalid" error="Error" scrollTooltip={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
