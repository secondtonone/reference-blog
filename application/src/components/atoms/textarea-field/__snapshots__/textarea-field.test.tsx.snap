// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots Textarea empty field 1`] = `
<span
  className="field-container__Container-sc-qv1os7-0 jOqnRO"
  data-test="field-container"
>
  <span>
    <textarea
      className="___STextarea_1vhq7_gg_ _size_xl_1vhq7_gg_ _state_normal_1vhq7_gg_"
      data-ui-name="Textarea"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Text"
      size="xl"
      style={
        Object {
          "--resize_s1bpz4": "none",
        }
      }
      value=""
    />
  </span>
</span>
`;

exports[`Match snapshots Textarea internal 1`] = `
<span
  className="field-container__Container-sc-qv1os7-0 trAty"
  data-test="field-container"
>
  <span>
    <textarea
      className="___STextarea_1vhq7_gg_ _size_xl_1vhq7_gg_ _state_normal_1vhq7_gg_"
      data-ui-name="Textarea"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      size="xl"
      style={
        Object {
          "--resize_s1bpz4": "none",
        }
      }
      value="Text"
    />
  </span>
</span>
`;

exports[`Match snapshots Textarea with error 1`] = `
<span
  className="field-container__Container-sc-qv1os7-0 jOqnRO"
  data-test="field-container"
>
  <span>
    <textarea
      className="___STextarea_1vhq7_gg_ _size_xl_1vhq7_gg_ _state_invalid_1vhq7_gg_"
      data-ui-name="Textarea"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      size="xl"
      style={
        Object {
          "--resize_s1bpz4": "none",
        }
      }
      value="Text"
    />
  </span>
</span>
`;

exports[`Match snapshots Textarea with value 1`] = `
<span
  className="field-container__Container-sc-qv1os7-0 jOqnRO"
  data-test="field-container"
>
  <span>
    <textarea
      className="___STextarea_1vhq7_gg_ _size_xl_1vhq7_gg_ _state_normal_1vhq7_gg_"
      data-ui-name="Textarea"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      size="xl"
      style={
        Object {
          "--resize_s1bpz4": "none",
        }
      }
      value="Text"
    />
  </span>
</span>
`;
